nf_core/download.py

```diff
@@ -1695,6 +1723,33 @@ def tidy_tags_and_branches(self):
                 self.retry_setup_local_repo(skip_confirm=True)
                 raise DownloadError(e) from e
 
+    # "Private" method to add the additional custom tags to the repository.
+    def __add_additional_tags(self) -> None:
+        if self.additional_tags:
+            self.ensure_git_user_config(f"nf-core download v{nf_core.__version__}", "core@nf-co.re")
```

Comment:
are we sure we want to use the nf-core-bot as the default user? I would prefer an error I think

Reply:
It doesn't need to be the `nf-core bot` we are using here as default, but an error for that reason is not a real option in my opinion.

After all, the downloaded bare repositories are pretty ephemeral and one would need to configure `git` properly to avoid. Admittedly, it is just two git commands that need to be run once and then never again, but I for example use `nf-core download` exclusively on Rackham, so I have Singularity available. But I am not doing development work on Rackham, so I haven't set up git there properly. The same goes for the func user that we use to prepare the deployments.

Lastly, [mind that it was actually the failing CI](https://github.com/nf-core/tools/pull/2938#issuecomment-2080034180) that prompted me to include this, so if we switch back to error out, we first have to adapt our Pytest CI 🫣.

Reply:
I agree with @MatthiasZepper on this one - this git action of adding a tag isn't really meaningful as a development step. This is purely an action to make running the pipeline easier, where the user behind the action is not visible. So I don't think that it matters what the account is, so I would prefer not to error.

That being said, I don't think that it needs to be a real account either? So maybe we can set it to `automated@tag.com` or some other fake email..?

Reply:
Yes, I have changed the default e-mail address. I presumed, it was a _sinkhole / do not reply_ address, but apparently not.